src/libcadet/model/parts/ConvectionDispersionOperator.cpp

```diff
@@ -185,29 +186,25 @@ bool ConvectionDispersionOperatorBase::configure(UnitOpIdx unitOpIdx, IParameter
  */
 bool ConvectionDispersionOperatorBase::notifyDiscontinuousSectionTransition(double t, unsigned int secIdx)
 {
-	double prevVelocity = static_cast<double>(_curVelocity);
-
 	// If we don't have cross section area, velocity is given by parameter
+	double dir_old = _dir;
 	if (_crossSection <= 0.0)
-		_curVelocity = getSectionDependentScalar(_velocity, secIdx);
+	{
+		double dir_new = static_cast<double>(getSectionDependentScalar(_velocity, secIdx));
+		_dir = (dir_new >= 0);
+	}
 	else if (!_velocity.empty())
 	{
-		if (secIdx > 0)
-		{
-			const double dir = static_cast<double>(getSectionDependentScalar(_velocity, secIdx - 1));
-			if (dir < 0.0)
-				prevVelocity *= -1.0;
-		}
-
 		// We have both cross section area and interstitial flow rate
 		// _curVelocity has already been set to the network flow rate in setFlowRates()
 		// the direction of the flow (i.e., sign of _curVelocity) is given by _velocity
-		const double dir = static_cast<double>(getSectionDependentScalar(_velocity, secIdx));
-		if (dir < 0.0)
-			_curVelocity *= -1.0;
+		double dir_new = static_cast<double>(getSectionDependentScalar(_velocity, secIdx));
+		_dir = (dir_new >= 0);
 	}
+	if (dir_old * _dir < 0.0)
+		_curVelocity *= -1.0;
```

Comment:
Is this working or should we do this instead:
```cpp
_curVelocity = abs(_curVelocity) * _dir;
```

Reply:
I think it works; it only changes the sign velocity iff dir_old != _dir; but I don't mind  changing it.

Reply:
You're probably right. When a discontinuous section transition occurs, the `ModelSystem` first calls `setFlowRates()`. Here, the velocity gets the sign of the previous `_dir`. Then, `notifyDiscontinuousSectionTransition()` is called. Here, we reverse the direction if necessary. So I agree, this should be correct.

However, the flipping must occur only in the `else if` branch above. Otherwise, this could happen:
Constant flow rate per section, transition from positive to negative. `_curVelocity` is set in the `if` branch and it is negative as it should be. We detect flow reversal and flip it again. `_curVelocity` ends up positive, which is wrong.